### PR TITLE
feat: runtime param

### PR DESCRIPTION
**Pull request by AbhishekA1509**
# Description
This PR adds runtime params in CD Material service and makes necessary changes in KeyValueTable to ensure migration of existing runtime params to a new design.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] Does this PR require documentation updates?
- [ ] I've updated documentation as required by this PR.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Shared/Components/KeyValueTable/KeyValueTable.types.ts [hunk 1]
```diff
@@ -97,18 +123,15 @@ export interface KeyValueTableProps<K extends string> {
...
     /**
      * The function to use to validate the value of the cell.
      * @param value - The value to validate.
      * @param key - The row key of the value.
+     * @param rowId - The id of the row.
      * @returns Return true if the value is valid, otherwise false
      * and set `showError` to `true` and provide errorMessages array to show error message.
      */
-    validationSchema?: (value: string, key: K) => boolean
+    validationSchema?: (value: string, key: K, rowId: string | number) => boolean
```
Comment (AbhishekA1509) [inline src/Shared/Components/KeyValueTable/KeyValueTable.types.ts:134]: Should i deprecate this prop?

File: src/Shared/Components/KeyValueTable/KeyValueTable.component.tsx [hunk 2]
```diff
@@ -85,81 +91,198 @@ export const KeyValueTable = <K extends string>({
...
+                    stringComparatorBySortOrder(a.data[sortBy].value, b.data[sortBy].value, sortOrder),
+                )
+                return sortedRows
+            })
+        }
+    }, [sortOrder])
+
+    useEffect(() => {
+        const firstRow = updatedRows?.[0]
+
```
Comment (AbhishekA1509) [inline src/Shared/Components/KeyValueTable/KeyValueTable.component.tsx:240]: Have removed null check from updatedRows, reminder to sync regarding that

File: src/Shared/Components/KeyValueTable/KeyValueTable.component.tsx [hunk 3]
```diff
@@ -85,81 +91,198 @@ export const KeyValueTable = <K extends string>({
...
         } as KeyValueRow<K>
+
+        return data
+    }
+
+    const handleAddNewRow = () => {
+        const data = getEmptyRow()
         const editedRows = [data, ...updatedRows]
 
+        const { id } = data
```
Comment (AbhishekA1509) [inline src/Shared/Components/KeyValueTable/KeyValueTable.component.tsx:204]: Not propagating onChange to parent on addition of row

File: src/Shared/Components/KeyValueTable/KeyValueTable.component.tsx [hunk 4]
```diff
@@ -334,12 +461,13 @@ export const KeyValueTable = <K extends string>({
                                     {!readOnly && (
                                         <button
```
Comment (AbhishekA1509) [inline src/Shared/Components/KeyValueTable/KeyValueTable.component.tsx:462]: Should we add check of row.data[key].disabled here? Reminder to sync with rohit

File: src/Shared/Components/KeyValueTable/KeyValueTable.types.ts (hunk 1 above)
Comment (eshankvaish) [reply src/Shared/Components/KeyValueTable/KeyValueTable.types.ts:134]: No

File: src/Shared/Components/KeyValueTable/KeyValueTable.component.tsx [hunk 5]
```diff
@@ -64,16 +68,20 @@ export const KeyValueTable = <K extends string>({
...
     // STATES
     const [updatedRows, setUpdatedRows] = useState<KeyValueRow<K>[]>(rows)
+    /** State to trigger useEffect to trigger autoFocus */
     const [newRowAdded, setNewRowAdded] = useState(false)
 
+    const isActionDisabled = readOnly || isAdditionNotAllowed
+
     /** Boolean determining if table has rows. */
     const hasRows = (!readOnly && !isAdditionNotAllowed) || !!updatedRows.length
+    const isFirstRowEmpty = !updatedRows[0]?.data[firstHeaderKey].value && !updatedRows[0]?.data[secondHeaderKey].value
```
Comment (eshankvaish) [inline src/Shared/Components/KeyValueTable/KeyValueTable.component.tsx:78]: @RohitRaj011 Pls confirm if the data[firstHeaderKey] can be null

**Review by eshankvaish — DISMISSED** [review]

File: src/Shared/Components/KeyValueTable/KeyValueTable.component.tsx (hunk 5 above)
Comment (RohitRaj011) [reply src/Shared/Components/KeyValueTable/KeyValueTable.component.tsx:78]: can't be null

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**